.github/workflows/pr-push.yml

```diff
@@ -0,0 +1,74 @@
+name: Pull Request
+on: pull_request
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
```

Comment:
TODO (if decide to run parallel jobs): reuse nodes in yml

Reply:
Previously we ran lint first, then headless, then the browser tests in parallel. Is that still doable? 

Also, yes, Firefox is 2x slower than Chrome unfortunately. Known issue.

Reply:
I think we should be able to do something like:
``` yaml
jobs:
  lint: ... # 1st
  headless:
   needs: lint # 2nd
  tests1:
    needs: headless # 3.1, parallel
  tests2:
    needs: headless # 3.2, parallel
```
Each job will require to repeat the "clone, install and build" steps though (which takes 1min at least).

But why would it be better than running all in parallel? Early exit?

Reply:
Lint is usually really quick, same with headless. The karma jobs were usually really slow.

Reply:
The lint step took 15 seconds, but set up step for lint took 45 seconds (no build needed after install).
In headless, tests took 18s, but setup took 1m 30s.
If we wait for headless and lint, we've added >2 min of delay.

If we run in series, that 1m 30s is reused in all steps (clone, install and build once, then do other steps - but other steps are done in series).
Ideally, I would've loved if the specific steps ran in parallel after common steps (not possible right now).

Reply:
Decisions decisions

Reply:
I think we should mirror what we originally had, even if it adds 2 minutes. The linting is usually what catches most issues for us, and there is no point in running karma if eslint fails.

Reply:
| Test  | Time |
| --- | --- |
| Lint | 1m 04s |
| Headless | 1m 47s |
| Karma Chrome | 2m 33s |
| Karma Firefox | 3m 25s |
| **Total** | 6m 16s |

**Summary**:

| Strategy |  Time |
| --- | --- |
| Parallel |  3m 23s |
| Travis like |  6m 16s |
| Series |  5m |
| Travis |  8-10m |

Reply:
I think we better run series... looking at the results in PR description

Reply:
Whatever you think it’s best. I don’t mind.